### PR TITLE
PICARD-2161: Add function to remove duplicate entries from multi-value variable

### PR DESCRIPTION
**Pull request by rdswift**
# Summary

* This is a…
    * [ ] Bug fix
    * [x] Feature addition
    * [ ] Refactoring
    * [ ] Minor / simple change (like a typo)
    * [ ] Other
* **Describe this change in 1-2 sentences**:  This adds a new scripting function `$uniquemulti()` to remove duplicate items from a multi-value variable.

# Problem

Duplicate items can be created in a multi-value variable when combining lists such as genres.  There is no easy way to remove duplicate entries.

* JIRA ticket (_optional_): PICARD-2161

# Solution

Add a new scripting function `$uniquemulti()` to remove duplicate entries.  Include an option to do case-sensitive comparisons.

# Action

Note to update the Picard documentation to include this new function.

### CODE REVIEW

**Comment by rdswift** [comment]
Would this be better if I used the `picard.util.uniquify()` function, and dropped the ability for case-insensitive comparisons?

**Comment by zas** [comment]
> Would this be better if I used the `picard.util.uniquify()` function, and dropped the ability for case-insensitive comparisons?

Nope, code looks good as is to me.

File: picard/script/functions.py [hunk 1]
```diff
@@ -1296,3 +1296,29 @@ def func_sortmulti(parser, multi, separator=MULTI_VALUED_JOINER):
...
+    $uniquemulti(a; A; B; b; a; b; A; B, cd,True)
+
+Result: A; B; a; b; cd
+"""
+))
+def func_uniquemulti(parser, multi, case_sensitive="", separator=MULTI_VALUED_JOINER):
+    multi_value = MultiValue(parser, multi, separator)
+    if not case_sensitive:
+        multi_value._multi = list({v.lower(): v for v in multi_value}.values())
+    return multi_value.separator.join(sorted(list(set(multi_value))))
```
Comment (zas) [inline picard/script/functions.py:1324]: `sorted(list(set(...)))` -> `sorted(set(...))`

**Review by zas — CHANGES_REQUESTED** [review]

**Comment by phw** [comment]
Could we get away with just $unique instead of $uniquemulti or would that be unclear?

For me I would always expect a function $unique to work on a list of things.

**Comment by rdswift** [comment]
> Could we get away with just $unique instead of $uniquemulti or would that be unclear?
> 
> For me I would always expect a function $unique to work on a list of things.

I'm okay with changing it to just `$unique`.  That actually makes more sense.

**Review by zas — APPROVED** [review]

**Review by phw — APPROVED** [review]

**Pull request merged**